components/Markets/ConditionalMarketOrderBook.tsx

```diff
@@ -10,107 +17,288 @@ export function ConditionalMarketOrderBook({
   orderBookObject: _OrderBook;
   setPriceFromOrderBook: (price: string) => void;
 }) {
-  let bids = orderBookObject?.failBidsArray;
-  let asks = orderBookObject?.failAsksArray;
-  let spreadString = orderBookObject?.failSpreadString;
+  if (!orderBookObject) return null;
+  const { colorScheme } = useMantineColorScheme();
+  const provider = useProvider();
+  const proposal = useProposal();
+  const [bids, setBids] = useState<any[][]>();
+  const [asks, setAsks] = useState<any[][]>();
+  const [wsConnected, setWsConnected] = useState<boolean>(false);
+  const [spreadString, setSpreadString] = useState<string>();
+  const [lastSlotUpdated, setLastSlotUpdated] = useState<number>(0);
+  const openBookProgram = new Program<OpenbookV2>(OPENBOOK_IDL, OPENBOOK_PROGRAM_ID, provider);
+
+  // On initialization
   if (isPassMarket) {
-    bids = orderBookObject?.passBidsArray;
-    asks = orderBookObject?.passAsksArray;
-    spreadString = orderBookObject?.passSpreadString;
+    if (!bids) {
+      setBids(orderBookObject.passBidsArray);
```

Comment:
Why couldn't you just do `const [bids, setBids] = useState(isPassMarket ? orderBookObject.passBidsArray : orderBookObject.failBidsArray)`? Would be a bit cleaner that way plus more usual for react code (where the initial value is usually just passed directly into the useState method